lib/services/role.go

```diff
@@ -1575,7 +1586,10 @@ func (set RoleSet) CheckAccessToServer(login string, s Server) error {
 	// one role in the role set to be granted access.
 	for _, role := range set {
 		matchNamespace, namespaceMessage := MatchNamespace(role.GetNamespaces(Allow), s.GetNamespace())
-		matchLabels, labelsMessage := MatchLabels(role.GetNodeLabels(Allow), s.GetAllLabels())
+		matchLabels, labelsMessage, err := MatchLabels(role.GetNodeLabels(Allow), s.GetAllLabels())
+		if err != nil {
+			return trace.CompareFailed(err.Error())
```

Comment:
Is `trace.CompareFailed` checked anywhere? If not, I would just `trace.Wrap(err)` here as well.